app/models/manageiq/providers/inventory.rb

```diff
@@ -10,11 +10,11 @@ class Inventory
     def self.build(ems, target)
       collector = collector_class_for(ems, target).new(ems, target)
       persister = persister_class_for(ems, target).new(ems, target)
-      new(
-        persister,
-        collector,
-        parser_classes_for(ems, target).map(&:new)
-      )
+      parsers   = parser_classes_for(ems, target).map do |parser_klass|
+        parser_klass.new(collector, persister)
+      end
+
+      new(persister, collector, parsers)
```

Comment:
The refactoring in this file isn't strictly related other than I wanted to use an attribute for https://github.com/ManageIQ/manageiq/pull/21460/files#diff-48cd14afd35edf466192c779dde0ae536dc7f2283d736008c001f39ab487e5b5R19 which wasn't set by initialize and I have no idea why setting these properties was delayed until right before parse was about to be called.